mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipArchivesLoaderTest.java

```diff
@@ -0,0 +1,298 @@
+        MailboxWithAnnotationsArchiveEntry resultMailbox = (MailboxWithAnnotationsArchiveEntry) mailArchiveIterator.next();
+        verifyMailboxArchiveEntry(mailArchiveIterator, expectedMailbox, resultMailbox, false);
+    }
+
+    @Test
+    void mailAccountIteratorFromArchiveWithOneMailboxAndTwoAnnotationShouldContainsOneMailbox() throws Exception {
+        MailboxSession session = mailboxManager.createSystemSession(USER);
+        createMailBoxWithMessage(session, MAILBOX_PATH_USER1_MAILBOX1);
+        mailboxManager.updateAnnotations(MAILBOX_PATH_USER1_MAILBOX1, session, WITH_ANNOTATION_1_AND_2);
+
+        ByteArrayOutputStream destination = new ByteArrayOutputStream(BUFFER_SIZE);
+        backup.backupAccount(USER1, destination);
+
+        InputStream source = new ByteArrayInputStream(destination.toByteArray());
+        MailArchiveIterator mailArchiveIterator = archiveLoader.load(source);
+        assertThat(mailArchiveIterator.hasNext()).isEqualTo(true);
+
+        MailboxWithAnnotationsArchiveEntry expectedMailbox = new MailboxWithAnnotationsArchiveEntry(MAILBOX_1_NAME, MAILBOX_ID_1.serialize(), WITH_ANNOTATION_1_AND_2);
+        MailboxWithAnnotationsArchiveEntry resultMailbox = (MailboxWithAnnotationsArchiveEntry) mailArchiveIterator.next();
+        verifyMailboxArchiveEntry(mailArchiveIterator, expectedMailbox, resultMailbox, false);
+    }
+
+    @Test
+    void mailAccountIteratorFromArchiveWithOneMailboxAndOneMessageShouldContainsOneMailboxAndOneMessage() throws Exception {
+        MailboxSession session = mailboxManager.createSystemSession(USER);
+        createMailBoxWithMessage(session, MAILBOX_PATH_USER1_MAILBOX1, MESSAGE_1);
+        mailboxManager.updateAnnotations(MAILBOX_PATH_USER1_MAILBOX1, session, NO_ANNOTATION);
+
+        ByteArrayOutputStream destination = new ByteArrayOutputStream(BUFFER_SIZE);
+        backup.backupAccount(USER1, destination);
+
+        InputStream source = new ByteArrayInputStream(destination.toByteArray());
+        MailArchiveIterator mailArchiveIterator = archiveLoader.load(source);
+        assertThat(mailArchiveIterator.hasNext()).isEqualTo(true);
+
+        MailboxWithAnnotationsArchiveEntry expectedMailbox = new MailboxWithAnnotationsArchiveEntry(MAILBOX_1_NAME, MAILBOX_ID_1.serialize(), NO_ANNOTATION);
+        MailboxWithAnnotationsArchiveEntry resultMailbox = (MailboxWithAnnotationsArchiveEntry) mailArchiveIterator.next();
+        verifyMailboxArchiveEntry(mailArchiveIterator, expectedMailbox, resultMailbox, true);
+
+        try (InputStream expectedContentStream = CONTENT_STREAM_1.newStream(0, -1)) {
+            MessageArchiveEntry expectedMessage = new MessageArchiveEntry(MESSAGE_ID_1.serialize(), MAILBOX_ID_1.serialize(),
+                SIZE_1, MESSAGE_1.getInternalDate(), flags1, expectedContentStream);
+
+            //---------TODO SEE WITH THE TEAM WHY THE FLAGS ARE NOT STORED or FETCHED, ONLY FOR DEBUG, TO REMOVE-----------//
```

Comment:
TODO spotted

Reply:
What's the issue BTW?

Reply:
The user flags are not retrieved/stored.
I persist a message with a flag "myFlag",
when i load it i get a message with a flag "\RECENT", but without the flag i set.